### PR TITLE
[WIP] Add geth+evmc tests

### PR DESCRIPTION
**Pull request by gballet**

### CODE REVIEW

**Comment by gballet** [comment]
@cdetrio @jwasinger I am downloading the go distrib and compiling `geth` each time, which is not efficient. Any input appreciated.

File: circle.yml [hunk 1]
```diff
@@ -97,6 +97,27 @@ defaults:
...
+    run:
+      name: "Build geth with EVMC support"
+      working_directory: ~/
+      command: |
+        mkdir -p go
+        mkdir -p goroot
+        export GOPATH=$HOME/go
+        export GOROOT=$HOME/goroot/go
+        curl https://dl.google.com/go/go1.11.3.linux-amd64.tar.gz > go1.11.3.linux-amd64.tar.gz
+        tar xfz go1.11.3.linux-amd64.tar.gz -C $HOME/goroot
```
[the remark this answers is not in the document]
Comment (jwasinger) [reply circle.yml]: You can mount it as a volume in the container

**Comment by axic** [comment]
This doesn't seem to work:
```
ok  	github.com/ethereum/go-ethereum/tests	0.135s [no tests to run]
```

**Comment by axic** [comment]
@gballet @chfast can you please have a look at this?

File: circle.yml [hunk 2]
```diff
@@ -97,6 +97,27 @@ defaults:
...
+        export PATH=$PATH:$GOROOT/bin:$GOPATH/bin
+        go get github.com/ethereum/go-ethereum
+        cd $GOPATH/src/github.com/ethereum/go-ethereum
+        git remote add ewasm http://github.com/ewasm/go-ethereum
+        git fetch ewasm
+        git checkout -b ewasm ewasm/ewasm
+        git submodule init
+        go build -o geth+evmc ./cmd/geth/...
+        go test ./tests/... -run TestState/stEWASMTests --vm.ewasm="$HOME/build/src/libhera.so"
```
Comment (axic) [inline circle.yml]: @gballet the bug is that `stEWASMTests` doesn't point to our version of the tests, but stock go-ethereum, in which case it is an empty directory.

Comment (gballet) [reply circle.yml]: that probably explains why it worked for me, ok thanks

**Comment by axic** [comment]
@gballet as you've mentioned the run time seems too fast. Is there a way to display the test cases executed?

**Comment by axic** [comment]
@gballet I think the testing here seems to work, but it crashes:
```
=== CONT  TestState/stEWASMTests/callDelegateOOG.json
terminate called after throwing an instance of 'hera::InvalidMemoryAccess'
  what():  Out of bounds (source) memory copy.
SIGABRT: abort
PC=0x7f962d57385b m=12 sigcode=18446744073709551610
```

**Comment by axic** [comment]
@gballet are you still intending to fix this?

File: circle.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (axic) [reply circle.yml]: At the bare minimum we should cache this 90M file next to aleth.

**Comment by gballet** [comment]
Don't think it makes sense to keep this open. Closing.

**Pull request closed without merging**